lib/policies/load-balancing.js

```diff
@@ -128,25 +130,21 @@ DCAwareRoundRobinPolicy.prototype.init = function (client, hosts, callback) {
   this.hosts = hosts;
   hosts.on('add', this._cleanHostCache.bind(this));
   hosts.on('remove', this._cleanHostCache.bind(this));
-  if (!this.localDc) {
-    //get the first alive local, it should be local on top
-    const hostsArray = hosts.values();
-    let dcHost;
-    for (let i = 0; i < hostsArray.length; i++) {
-      const h = hostsArray[i];
-      if (h.datacenter) {
-        this.localDc = h.datacenter;
-        dcHost = h;
-        break;
-      }
-    }
-    //this should never happen but it does not hurt
-    if (!this.localDc) {
-      return callback(new errors.DriverInternalError('Local datacenter could not be determined'));
+
+  if (client && client.options) {
+    if (this.localDc && !client.options.localDataCenter) {
+      this.client.log('warning', 'Local data center \'' + this.localDc + '\' was provided as an argument to' +
+        ' DCAwareRoundRobinPolicy. It is more preferable to specify the local data center using \'localDataCenter\'' +
+        ' in Client options instead.');
     }
-    client.log('warning', 'No local Data Center was provided with DCAwareRoundRobinPolicy' +
-      '.  Using discovered DC \'' + this.localDc + '\' from host ' + dcHost.address +
-      '.  Future releases will require local DC to be specified.');
+
+    // If localDc is unset, use value set in client options.
+    this.localDc = this.localDc || client.options.localDataCenter;
+  }
+
+  if (!this.localDc) {
+    return callback(new errors.DriverInternalError('\'localDataCenter\' is not defined in Client options and also was' +
```

Comment:
We can use `ArgumentError` as this can only be specified by the user

```suggestion
    return callback(new errors.ArgumentError('\'localDataCenter\' is not defined in Client options and also was' +
```